src/test/ui/lint/liveness-unused.stderr

```diff
@@ -52,7 +52,7 @@ warning: value assigned to `x` is never read
   --> $DIR/liveness-unused.rs:42:5
    |
 LL |     x += 4;
-   |     ^
+   |     ^    ^
```

Comment:
This seems a bit surprising to me. I feel like highlighting multiple things makes sense when they are a homogeneous list, but here they are two distinct things, so I would expect labels. I can't however come up with obviously *good* labels. Probably label `4` with something like "value" and `x` with ... I don't know. There is also a risk here when the expression on the RHS spans multiple lines, in which case the underline looks pretty bad usually:

```
let x = foo(
    1,
    2,
);
```

cc @estebank -- thoughts?

Reply:
Hello,
An expression or statement boundary check to see if it spans more than one line would be needed for multiple line layouts. In this isolated case, we are looking for the semi-colon and if the backtrack has lines greater than one, use another warning marker layout.

An argument could be made to run that snippet through a **format code** cycle just to provide more streamlined feedback. Your example would collapse to `let x = foo(1, 2);` and it would lead to less visual edge cases for providing warnings. Not sure about how to carve it out or the nature of retaining original line numbering. The flattening could just be for display purposes maybe.

For labeling the pieces, you may just want to use a different character to point out the LHS from the RHS. so **`<^^`** on LHS and **`>^^`** on RHS to distinguish binding from value. No sure if all this is better in the original ticket, I will transplant if so. Thank you. Good day.